### PR TITLE
Add `opm` docs

### PR DESCRIPTION
**Pull request by kevinrizza**
* Add documentation that defines the input and output of all public opm
commands

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,82 +17,75 @@ And libraries:
...
-    │   └── servicemonitor.crd.yaml
-    ├── 0.22.2
-    │   ├── alertmanager.crd.yaml
-    │   ├── prometheus.crd.yaml
-    │   ├── prometheusoperator.0.22.2.clusterserviceversion.yaml
-    │   ├── prometheusrule.crd.yaml
-    │   └── servicemonitor.crd.yaml
-    └── prometheus.package.yaml
+
+The operator-sdk also provides tooling to generate the metadata, Dockerfile, and even build the container image itself. Some basic documentation on how this works is located in [this repository's documentation](docs/design/operator-bundle.md#Operator-SDK-CLI).
```
Comment (dinhxuanvu) [inline README.md]: Note here that "operator-sdk" reference is being removed on the bundle content validation PR. So we may need to update this later after the PR merged.

File: README.md [hunk 2]
```diff
@@ -16,82 +17,75 @@ And libraries:
...
  * `pkg/sqlite` - providing interfaces for building sqlite manifest databases from `ConfigMap`s or directories, and for querying an existing sqlite database.
+ * `pkg/lib` - providing external interfaces for interacting with this project as an api that defines a set of standards for operator bundles and indexes.
+ * `pkg/containertools` - providing an interface to interact with and shell out to common container tooling binaries (if installed on the environment)
 
 # Manifest format
 
 
-We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle is typically just a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV, though additional objects may be included.
+We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle typically includes a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV in its manifest directory, though additional objects may be included. It also includes an annotations file in its metadata folder which defines some higher level aggregate data that helps describe the format and package information about how the bundle should be added into an index of bundles.
```
[the remark this answers is not in the document]
Comment (dinhxuanvu) [reply README.md]: Also, do you think we should have a section to explain the concept of "an index" or what does it mean? Is it a file or an image or something else? We use the reference "index? in quite a few places here so maybe it deserves its own section perhaps?

[the remark this answers is not in the document]
Comment (kevinrizza) [reply README.md]: good point about channels.

but i sort of feel like that's what the purpose of this whole section is: https://github.com/operator-framework/operator-registry/blob/2066c95d074c1e4034129eea2651742776099cd4/README.md#building-an-index-of-operators-using-opm

maybe it needs a one line definition though? i can add that.

File: README.md (hunk 1 above)
Comment (kevinrizza) [reply README.md]: sure, that sounds good. I think we can just remove this and then update it once the sdk pushes their own doc we can link to.

**Comment by kevinrizza** [comment]
@dinhxuanvu updated based on your feedback.

File: README.md [hunk 3]
```diff
@@ -16,82 +17,75 @@ And libraries:
...
- * If a ClusterServiceVersion `replaces` another, both the old and the new must exist in the package.
 
 Bundle directories are identified solely by the fact that they contain a ClusterServiceVersion, which provides an amount of freedom for layout of manifests.
 
-It's recommended to follow a layout that makes it clear which bundles are part of which package, as in [manifests](manifests):
+Check out the [operator bundle design](docs/design/operator-bundle.md) for more detail on the bundle format.
 
+# Bundle images
+
+Using [OCI spec](https://github.com/opencontainers/image-spec/blob/master/spec.md) container images as a method of storing the manifest and metadata contents of individual bundles, `opm` interacts directly with these images to generate and incrementally update the database. Once you have your [manifests defined](https://operator-framework.github.io/olm-book/docs/packaging-an-operator.html#writing-your-operator-manifests) and have created a directory in the format defined above, building the image is as simple as defining a [Dockerfile](docs/design/operator-bundle.md#Bundle-Dockerfile) and building that image:
```
Comment (benluddy) [inline README.md]: I'm unclear on how flexible the directory structure is intended to be. Is a distinction made between a bundle directory structure described above and the layout of files in a bundle image?

I see the statement "bundle directories are identified solely by the fact that they contain a ClusterServiceVersion, which provides an amount of freedom for layout of manifests," and I'm not sure how that aligns with the proscribed metadata and manifests directories mentioned above.

File: README.md [hunk 4]
```diff
@@ -16,82 +17,75 @@ And libraries:
...
  * `pkg/sqlite` - providing interfaces for building sqlite manifest databases from `ConfigMap`s or directories, and for querying an existing sqlite database.
+ * `pkg/lib` - providing external interfaces for interacting with this project as an api that defines a set of standards for operator bundles and indexes.
+ * `pkg/containertools` - providing an interface to interact with and shell out to common container tooling binaries (if installed on the environment)
 
 # Manifest format
 
 
-We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle is typically just a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV, though additional objects may be included.
+We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle typically includes a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV in its manifest directory, though additional objects may be included. It also includes an annotations file in its metadata folder which defines some higher level aggregate data that helps to describe the format and package information about how the bundle should be added into an index of bundles.
```
[the remark this answers is not in the document]
Comment (kevinrizza) [reply README.md]: CRDs are not mandatory, actually. It minimally includes just the CSV.

File: README.md (hunk 3 above)
Comment (kevinrizza) [reply README.md]: The only directory we care about is the one on the image. You could, in theory, write your own Dockerfile that pulls files from lots of places to generate the directory structure in place. The flexibility is that the manifest directory can have certain other kube manifests optionally (currently that list includes rbac files and some prometheus stuff aside from the CSVs and CRDs). In future iterations this list can (and will be) extended. But all of that does need to live in the `manifests/` subdirectory on your image.

Right now `metadata/` has just the `annotations.yaml` file, but we are intending for this to be extended in the same fashion.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dinhxuanvu) [reply README.md]: I think this is good for now. We can continue to improve the doc as we go. We need a starting point first.

**Pull request merged**